its/plugin/src/test/java/org/sonarsource/iac/TestBase.java

```diff
@@ -112,4 +121,31 @@ protected static WsClient newWsClient() {
       .url(ORCHESTRATOR.getServer().getUrl())
       .build());
   }
+
+  public static void executeBuildWithExpectedWarnings(Orchestrator orchestrator, SonarScanner build) {
+    BuildResult result = orchestrator.executeBuild(build);
+    assertAnalyzerLogs(result.getLogs());
+  }
+
+  private static void assertAnalyzerLogs(String logs) {
+    List<String> lines = Arrays.asList(logs.split("[\r\n]+"));
+
+    assertThat(lines.size()).isBetween(25, 150);
+
+    List<String> unexpectedLogs = lines.stream()
+      .filter(line -> !line.startsWith("INFO: "))
+      .filter(line -> !line.startsWith("WARN: SonarQube scanners will require Java 11+ starting on next version"))
+      .filter(line -> !line.startsWith("WARN: The sonar.modules is a deprecated property and should not be used anymore"))
+      .filter(line -> !line.startsWith("WARNING: An illegal reflective access operation has occurred"))
+      .filter(line -> !line.startsWith("WARNING: Illegal reflective access"))
+      .filter(line -> !line.startsWith("WARNING: Please consider reporting this to the maintainers"))
+      .filter(line -> !line.startsWith("WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations"))
+      .filter(line -> !line.startsWith("WARNING: All illegal access operations will be denied in a future release"))
+      .filter(line -> !line
+        .startsWith("WARN: The property 'sonar.login' is deprecated and will be removed in the future. Please use the 'sonar.token' property instead when passing a token."))
+      .filter(line -> !line.startsWith("Picked up JAVA_TOOL_OPTIONS:"))
+      .collect(Collectors.toList());
+
+    assertThat(unexpectedLogs).isEmpty();
+  }
```

Comment:
I know this is copied from PHP, but I think we can do it smarter with a set of strings instead of chained filters